packages/arb-token-bridge-ui/src/state/app/utils.ts

```diff
@@ -307,15 +305,34 @@ export const isDepositReadyToRedeem = (tx: MergedTransaction) => {
   return isDeposit(tx) && tx.depositStatus === DepositStatus.L2_FAILURE
 }
 
-export const getStandardizedTimestamp = (date: string | BigNumber) => {
+export const normalizeTimestamp = (date: number | string | BigNumber) => {
   // because we get timestamps in different formats from subgraph/event-logs/useTxn hook, we need 1 standard format.
+  const TIMESTAMP_LENGTH = 13
+  let timestamp
+
   if (typeof date === 'string') {
-    if (isNaN(Number(date))) return dayjs(new Date(date)).unix() // for ISOstring type of dates -> dayjs timestamp
-    return Number(date) // for timestamp type of date -> dayjs timestamp
+    timestamp = isNaN(Number(date))
+      ? dayjs(new Date(date)).unix() // for ISOstring type of dates -> dayjs timestamp
+      : Number(date) // for timestamp type of date -> dayjs timestamp
+  } else if (typeof date !== 'number') {
+    // BigNumber
+    timestamp = date.toNumber()
+  }
+
+  const timestampString = String(timestamp)
+
+  if (timestampString.length === TIMESTAMP_LENGTH) {
+    // correct timestamp length
+    return Math.floor(Number(timestampString))
+  }
+
+  if (timestampString.length < TIMESTAMP_LENGTH) {
+    // add zeros at the end until correct timestamp length
+    return Math.floor(Number(timestampString.padEnd(TIMESTAMP_LENGTH, '0')))
```

Comment:
it doesn't look like it needs `Math.floor` based on what it's doing here

Reply:
Right we don't divide, I will remove